### ISSUE TITLE
Fixing archive / delete alert message sentence structure

### ISSUE THREAD

**Issue by chiragraman** [nylas/N1#1921]
Hello,

This is an extremely minor UX issue. The user-facing message for the delete/archive action reads as 'Added <label> to <number of threads> threads', which is a little awkward. While this makes sense from the perspective of 'Added the label 'Trash' to 1 thread', the paradigm most users would be familiar with treats these labels conceptually as folders, with 'Moved to Trash' or 'Moved to Archive' being the standard message structure. The trash can hover text is indeed 'Move to Trash' too. So the alert label ought to match that structure. 

Further, the source code treats these labels as types of Categories, so conceptually, the action that makes sense is adding to, or removing things from categories, as opposed to the reverse.  

Thanks!

##### Are there any related issues?
None that I could find.

##### What operating system are you using?
OSX Yosemite 10.10.5

##### What version of N1 are you using?
Version 0.4.19-d41e72c

--

**Bug?**
##### Do you have any third-party plugins installed?
Nope

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
Nope

##### Is the issue reproducible with a particular attachment, message, signature, etc?
Yes, archive/delete any message.

**Comment by bengotow** [comment]
Hey! This is great feedback—definitely something we overlooked. We'll get this fixed in the next release.

**Comment by chiragraman** [comment]
Thanks Ben! I am in the process of coming up with a fix on a branch on my
fork. If you'd like I can issue a PR within the next few days.  
  
Sent from [Nylas N1](https://nylas.com/n1?ref=n1), the extensible, open source
mail client.  

On Apr 13 2016, at 1:58 pm, Ben Gotow &lt;notifications@github.com&gt; wrote:  

> Hey! This is great feedback—definitely something we overlooked. We'll get
this fixed in the next release.

>

> —  
You are receiving this because you authored the thread.  
Reply to this email directly or [view it on GitHub](https://github.com/nylas/N
1/issues/1921#issuecomment-209569375)![](https://github.com/notifications/beac
on/AB3E3fPi5a1jsz0qOVUCTk4j-mAIS-dXks5p3S7OgaJpZM4IDvad.gif)

**Comment by bengotow** [comment]
Hey! It looks like the most recent version says "Moving to Archive..." in the activity area in the bottom left, and "Moved 1 thread to Archive" in the alert. I think we can probably make the alert shorter (no reason to say the number of threads when it's just one), but I'm not seeing the "Added the label 'Trash' to one thread."

[Edit: It looks like this is implemented nicely in the `ChangeFolderTask`, and it's the `ChangeLabelsTask` for Gmail that is behaving the way you describe.]

![image](https://cloud.githubusercontent.com/assets/1037212/14503879/d8524216-0166-11e6-9c61-33f61bee1384.png)

**Comment by bengotow** [comment]
Sounds good—would love to see a PR for this 👍

**Comment by chiragraman** [comment]
Indeed, it is the change labels tasks line 48:  
  

```  
  return `Added ${this.labelsToAdd[0].displayName} to ${this.threads.length}
${type}`;  
```  
  
On my fork I was modifying it to match the behavior you just described for the
folders so that the UX is consistent across workflows.

  

Should have it soon!

  

Also, thanks for all the work on this, I'm on the whole loving it as a user :)

  

Best,

Chirag

  
Sent from [Nylas N1](https://nylas.com/n1?ref=n1), the extensible, open source
mail client.  

On Apr 13 2016, at 2:02 pm, Ben Gotow &lt;notifications@github.com&gt; wrote:  

> Sounds good—would love to see a PR for this ![](images/composer-
emoji/apple/1f44d.png)

>

> —  
You are receiving this because you authored the thread.  
Reply to this email directly or [view it on GitHub](https://github.com/nylas/N
1/issues/1921#issuecomment-209570724)![](https://github.com/notifications/beac
on/AB3E3UQetWUmW7zfLU6a2NzwzuSdfImEks5p3S-ygaJpZM4IDvad.gif)

**Comment by bengotow** [comment]
Hey! Any update on this? We saw a few more complaints about this come in to support@nylas. Would love to have a PR!

**Comment by bengotow** [comment]
(We're working on this internally, should be included in the next release)

**Comment by bengotow** [comment]
Hey! Thanks for reporting this—we shipped `0.4.45` this afternoon which should resolve this issue. It should ship via the autoupdater in the next 24 hours, or you can download it directly from the Releases page. I'm going to ahead and close this—if you continue seeing the issue, feel free to comment and I'll re-open it!
